### PR TITLE
Update versioning to 1.20.0 pre-release.

### PR DESCRIPTION
**Pull request by jhoole**
I only updated where I had previously changed from 1.19.0 pre-release
to 1.19.0. Places where I had updated from 1.18.0 to 1.19.0 I have
left at 1.19.0.

Compared changes from #12584.

### CODE REVIEW

**Comment by bradcray** [comment]
This line needs to be updated from 19 to 20:

https://github.com/chapel-lang/chapel/blob/master/compiler/main/version_num.h#L24

I'm also getting a spacing problem when I build the compiler and run `chpl --version` but am not seeing why yet.  Are you familiar with how to build and run it to try it manually?  The tests that check that the version is correct will need to have their expected output updated as well:

https://github.com/chapel-lang/chapel/blob/a0ddf07494c974efd2093b157d19cc1318352d25/test/compflags/bradc/printstuff/version.goodstart

I'm looking at PR #11111 for reference.

For those looking at this PR and wondering why doc/rst/conf.py wasn't updated, it's because I already did it in PR #12632

File: compiler/main/version.cpp [hunk 1]
```diff
@@ -35,7 +35,7 @@ void
 get_version(char *v) {
   v += sprintf(v, "%d.%s.%s", MAJOR_VERSION, MINOR_VERSION, UPDATE_VERSION);
   if (strcmp(BUILD_VERSION, "0") != 0 || developer)
-    sprintf(v, ".%s", BUILD_VERSION);
+    sprintf(v, "pre-release (%s)", BUILD_VERSION);
```
Comment (bradcray) [inline compiler/main/version.cpp]: I think this line probably needs the space?

**Comment by jhoole** [comment]
chpl --version looks good and versioning printstuff tests are passing.
```
MacBook-Pro:chapel jhoole$ chpl --version
chpl version 1.20.0 pre-release (1041e5dece)
Copyright (c) 2004-2019, Cray Inc.  (See LICENSE file for more details)
```

**Comment by bradcray** [comment]
This is looking OK to me, comparing against #11111 and doing some spot checks on a copy of the branch.

**Comment by jhoole** [comment]
@bradcray, is this alright to merge?

**Comment by bradcray** [comment]
Yes, sorry if I wasn't clear enough about that.

**Pull request merged**